core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -0,0 +1,88 @@
+module Spree
+  module UserAddressBook
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress" do
+        def find_by_address_values(address_attributes)
+          # finds by value, not id
+          # since Address is immutable, we treat it like a value object rather than an entity.
+          # If your database does not do case sensitive matching on varchar types, then you'll
+          # have trouble trying to correct an address field's case
+          self.joins(:address).readonly(false).where(
+            spree_addresses: address_attributes.stringify_keys.except(*Address::EQUALITY_IRRELEVANT_ATTRS)
+          ).first
+        end
+      end
+
+      has_many :addresses, through: :user_addresses
+
+      has_one :default_user_address, -> { where default: true}, foreign_key: "user_id", class_name: 'Spree::UserAddress'
+      has_one :default_address, through: :default_user_address, source: :address
+
+      # bill_address is only minimally used now, but we can't get rid of it without a major version release
+      belongs_to :bill_address, class_name: 'Spree::Address'
+
+      def default_address
+        super
```

Comment:
It demonstrates the hurry I was in trying to put this PR together. ;-) I'll remove